### PR TITLE
Phosphor Queue docs update

### PR DESCRIPTION
**Pull request by gochoam**
I broke the example up into smaller chunks and grouped similar methods, also added a brief description about what they do.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,68 +90,135 @@ Usage Examples
...
+queue = require('phosphor-queue')
+```
 
-q.push(42);
-q.push(43);
-q.push(44);
-q.push(45);
+This imports the `Queue()` constructor to create new queue. Once you create a
+new queue it's straightforward to check some basic properties: the front and
+back elements, the queue size and whether it is empty or not:
```
Comment (sccolbert) [inline README.md]: `s/whether it is empty or not/whether or not it is empty`

File: README.md [hunk 2]
```diff
@@ -91,68 +90,135 @@ Usage Examples
...
+> q.size;
+4
+> q.empty;
+false
+> 
+```
 
-q.some(v => v < 40);  // false
-q.some(v => v > 44);  // true
+The `pop()` method removes the back item and `push()` inserts a new front
```
Comment (sccolbert) [inline README.md]: `pop` removes the front, `push` adds to the back.

File: README.md [hunk 3]
```diff
@@ -91,68 +90,135 @@ Usage Examples
...
+```node
+> q.remove(42);
+true
+> q.remove(19);
+false
+> q.removeAll(44);
+2
+> q.removeAll(19);
+0
+> q.toArray(19);
```
Comment (sccolbert) [inline README.md]: `toArray` takes no arguments

File: README.md [hunk 4]
```diff
@@ -91,68 +90,135 @@ Usage Examples
...
+[ 43, 45 ]
+
+> q.clear();
+> q.size
+0
+```
 
-q.filter(v => v < 40);  // []
-q.filter(v => v > 44);  // [45]
+The `phosphor-queue` module also provides methods to check whether the elements
```
Comment (sccolbert) [inline README.md]: s/The `phosphor-queue` module/A `Queue` object

File: README.md [hunk 5]
```diff
@@ -91,68 +90,135 @@ Usage Examples
...
+in the queue pass a boolean-valued predicate function. These methods are :
```
Comment (sccolbert) [inline README.md]: s/are :/are:

File: README.md [hunk 6]
```diff
@@ -91,68 +90,135 @@ Usage Examples
...
+false
+
+> q.filter(v => v < 40);
+[]
+> q.filter(v => v > 44);
+[ 45 ]
+```
 
-q.map(v => v * 2);  // [84, 86, 88, 90]
+Two additional methods of PhosphorJS queue that provide extra flexibility are
```
Comment (sccolbert) [inline README.md]: s/of PhosphorJS queue/of the queue

**Comment by sccolbert** [comment]
@gochoam Made some comments

**Comment by gochoam** [comment]
@sccolbert The file now includes the suggested changes.

**Pull request merged**